scheduler/test/e2e/scheduler_test.go

```diff
@@ -36,118 +43,165 @@ var _ = Describe("[E2E] KeptnScheduler", Ordered, func() {
 			AppAnnotation:      "myapp",
 		}
 
-		pause := imageutils.GetPauseImageName()
-		var (
-			workloadinstance *testv1alpha2.KeptnWorkloadInstance
-			pod              *apiv1.Pod
-		)
+		pod := &apiv1.Pod{}
+
 		BeforeEach(func() {
 			DeferCleanup(func() {
 				err := k8sClient.Delete(ctx, pod)
 				logErrorIfPresent(err)
 			})
-
-			//create a test Pod
-			name := names.SimpleNameGenerator.GenerateName("my-testpod-")
-
-			pod = WithContainer(st.MakePod().
-				Namespace("default").
-				Name(name).
-				Req(map[apiv1.ResourceName]string{apiv1.ResourceMemory: "50"}).
-				ZeroTerminationGracePeriod().
-				Obj(), pause)
-			pod.Annotations = annotations
-
-			err := k8sClient.Create(ctx, pod)
-			Expect(ignoreAlreadyExists(err)).NotTo(HaveOccurred(), "could not add pod")
+			*pod = initPod(*pod, annotations, nil, KeptnScheduler)
 		})
 
 		Context("a new Pod", func() {
 
 			It(" should stay pending if no workload instance is available", func() {
-
-				newPod := &apiv1.Pod{}
-				Eventually(func() error {
-					err := k8sClient.Get(ctx, types.NamespacedName{Namespace: pod.Namespace, Name: pod.Name}, newPod)
-					return err
-				}).Should(Succeed())
-
-				Expect(newPod.Status.Phase).To(Equal(apiv1.PodPending))
-
+				checkPending(pod)
 			})
 
 			It(" should be scheduled when workload instance pre-evaluation checks are done", func() {
+				checkWorkload("myapp-myworkload-1.0.0", *pod, "Succeeded")
+			})
+		})
+	})
 
-				workloadinstance = initWorkloadInstance()
+	Describe("If NOT annotated or labeled for keptn-scheduler", func() {
+		pod := &apiv1.Pod{}
+		BeforeEach(func() {
+			DeferCleanup(func() {
+				err := k8sClient.Delete(ctx, pod)
+				logErrorIfPresent(err)
+			})
+			*pod = initPod(*pod, nil, nil, "default")
+		})
 
-				err := k8sClient.Create(ctx, workloadinstance)
-				Expect(ignoreAlreadyExists(err)).To(BeNil())
+		Context("a new Pod", func() {
 
-				Eventually(func() error {
-					err := k8sClient.Get(ctx, types.NamespacedName{Namespace: pod.Namespace, Name: "myapp-myworkload-1.0.0"}, workloadinstance)
-					return err
-				}).Should(Succeed())
-				workloadinstance.Status.PreDeploymentEvaluationStatus = "Succeeded"
-				err = k8sClient.Status().Update(ctx, workloadinstance)
+			It(" should be immediately scheduled", func() {
 
-				Expect(err).To(BeNil())
 				Eventually(func() error {
-					return podRunning(pod.Namespace, pod.Name)
-				}).Should(Succeed())
-
-				err = k8sClient.Delete(ctx, workloadinstance)
-				Expect(err).NotTo(HaveOccurred(), "could not remove workloadinstance")
+					return podScheduled(pod.Namespace, pod.Name)
+				}).WithTimeout(time.Second * 20).WithPolling(3 * time.Second).Should(Succeed())
```

Comment:
Should we be on the safe side here to avoid flakyness? 
```suggestion
				}).WithTimeout(time.Second * 60).WithPolling(3 * time.Second).Should(Succeed())
```